src/index.ts

```diff
@@ -1,93 +1,154 @@
-import * as core from '@actions/core';
-import * as github from '@actions/github'
-import { LinearClient } from '@linear/sdk'
+import * as core from "@actions/core";
+import * as github from "@actions/github";
+import { LinearClient } from "@linear/sdk";
 import { isNil } from "lodash";
 
 const PR_TITLE_UPDATE_KEYWORD = "x";
 
 type PullRequest = {
-    number: number;
-    title: string;
-    head: {
-        ref: string;
-        repo: {
-            name: string;
-            owner: {
-                login: string;
-            }
-        } | null,
-    },
-}
+  number: number;
+  title: string;
+  body: string | null;
+  head: {
+    ref: string;
+    repo: {
+      name: string;
+      owner: {
+        login: string;
+      };
+    } | null;
+  };
+};
 
-export const getIssueIdFromBranchName = (branchName: string): string | null => {
-    const match = branchName.match(/^feature\/([a-z]+\-\d+)\-/);
-    if (isNil(match)) return null;
-    const branchStoryId = match[1];
-    return branchStoryId.toUpperCase();
-}
+export const getLinearIssueIds = (pullRequest: PullRequest) => {
+  const issueIds: string[] = [];
+
+  // From the branch name
+  const branchName = pullRequest.head.ref;
+  let match = branchName.match(/^([a-z]+\-\d+)\-/);
+  if (!isNil(match)) issueIds.push(match[1].toUpperCase());
+
+  // From the PR body
+  const body = pullRequest.body ?? "";
+  const regexp = /Fixes ([a-z]+\-\d+)|Resolves ([a-z]+\-\d+)/gi;
+  const matches = [...body.matchAll(regexp)];
+
+  matches.map((match: string[]) =>
```

Comment:
This feels a little more readable and extensible:

```
const bodySearchRegexes = [
    /fixes ([a-z]+\-\d+)/gi,
    /resolves ([a-z]+\-\d+)/gi
]
const issueIdsFromBody = bodySearchRegexes.flatMap(regex => {
    const matches = [...body.matchAll(regex)];
    return matches.map(match => {
        const captureMatch = match[1];
        return captureMatch.toUpperCase();
    });
});
```

Reply:
What about this? https://github.com/perdoo/linear-pr-title-action/pull/2/commits/1e350c5dd54ca3b57d4206a7bcdb914b39f5e0ae

It's more readable, at least for me.